test/StakeManager.test.ts

```diff
@@ -0,0 +1,378 @@
+    })
+  }
+
+  function testCanPenalize (): void {
+    it('should allow to penalize hub', async function () {
+      const beneficiaryBalanceTracker = await balance.tracker(nonOwner)
+      const stakeManagerBalanceTracker = await balance.tracker(stakeManager.address)
+      const remainingStake = new BN(100)
+      const penaltyAmount = initialStake.sub(remainingStake)
+      const { logs } = await stakeManager.penalizeRegistree(registree, nonOwner, penaltyAmount, {
+        from: anyRelayHub
+      })
+      const expectedReward = penaltyAmount.divn(2)
+      expectEvent.inLogs(logs, 'StakePenalized', {
+        registree,
+        beneficiary: nonOwner,
+        reward: expectedReward
+      })
+      const relayOwnerGain = await beneficiaryBalanceTracker.delta()
+      const stakeManagerLoss = await stakeManagerBalanceTracker.delta()
+      expect(relayOwnerGain).to.be.bignumber.equal(expectedReward)
+      expect(stakeManagerLoss).to.be.bignumber.equal(penaltyAmount.neg())
+
+      // @ts-ignore
+      const { stake: actualStake, unstakeDelay: actualUnstakeDelay, owner: actualOwner } =
+        await stakeManager.stakes(registree)
+      expect(actualOwner).to.equal(owner)
+      expect(actualStake).to.be.bignumber.equal(remainingStake)
+      expect(actualUnstakeDelay).to.be.bignumber.equal(initialUnstakeDelay)
+    })
+  }
+
+  describe('with no stake for relay server', function () {
+    beforeEach(async function () {
+      stakeManager = await StakeManager.new()
+    })
+
+    testStakeNotValid()
+
+    it('should not allow not owner to schedule unlock', async function () {
+      await expectRevert(
+        stakeManager.unlockStake(nonOwner, { from: owner }),
+        'not owner'
+      )
+    })
+
+    it('registrees cannot stake for themselves', async function () {
+      await expectRevert(
+        stakeManager.stakeForAddress(registree, initialUnstakeDelay, {
+          value: initialStake,
+          from: registree
+        }),
+        'registree cannot stake for itself'
+      )
+    })
+
+    testCanStake(registree)
+  })
+
+  describe('with stake deposited for relay server', function () {
+    beforeEach(async function () {
+      stakeManager = await StakeManager.new()
+      await stakeManager.stakeForAddress(registree, initialUnstakeDelay, {
+        value: initialStake,
+        from: owner
+      })
+    })
+
+    it('should not allow to penalize hub', async function () {
+      await expectRevert(
+        stakeManager.penalizeRegistree(registree, nonOwner, initialStake, { from: anyRelayHub }),
+        'hub not authorized'
+      )
+    })
+
+    it('should allow querying registree\'s stake', async function () {
+      // @ts-ignore (typechain does not declare names or iterator for return types)
+      const { stake: actualStake, unstakeDelay: actualUnstakeDelay, owner: actualOwner } =
+        await stakeManager.stakes(registree)
+      expect(actualOwner).to.equal(owner)
```

Comment:
I prefer doing something like `assert.deepEqual( {stake, unstakeDelay, owner}, {stake:initialStake, unstakeDelay: initialUnstakeDelay, owner: intialOwner })`
this way, you have a single error on all (it is a little tricky with bignumbers, though..)

Reply:
The single error for 3 variables is not a good thing in any univers